src/main/java/org/sonar/plugins/checkstyle/metadata/CheckstyleMetadata.java

```diff
@@ -95,35 +99,53 @@ public void createRulesWithMetadata() {
                 new DefaultDebtRemediationFunction(DebtRemediationFunction.Type.CONSTANT_ISSUE,
                         null, "0d 0h 5min");
 
-        for (String checkName : newChecks) {
-            final ModuleDetails moduleDetails = loadMeta(checkName);
-            if (moduleDetails != null) {
-                final SonarRulePropertyLoader.AdditionalRuleProperties additionalDetails =
-                        additionalRuleData.get(checkName);
-                final RulesDefinition.NewRule rule =
-                        repository.createRule(moduleDetails.getFullQualifiedName());
-                rule.setHtmlDescription(moduleDetails.getDescription())
-                        .setName(convertName(moduleDetails.getName()))
-                        .setInternalKey(convertInternalKey(moduleDetails))
-                        .setDebtRemediationFunction(debtRemediationFunction)
-                        .setSeverity("MINOR")
-                        .setStatus(RuleStatus.READY);
-                final String tag = getRuleTag(moduleDetails.getFullQualifiedName(),
-                        additionalDetails);
-                if (tag != null) {
-                    rule.setTags(tag);
-                }
-                if (isTemplateRule(moduleDetails.getFullQualifiedName())) {
-                    rule.setTemplate(true);
-                }
+        metadataRepo.keySet().stream()
+                .filter(check -> { //NOSONAR
+                    return !existingChecks.contains(check)
+                            && metadataRepo.get(check).getModuleType() == ModuleType.CHECK
+                            // these checks are not available in checkstyle 8.35, these conditions
+                            // should be removed when upgrading to 8.36
+                            && !check.contains("RecordTypeParameterNameCheck")
+                            && !check.contains("PatternVariableNameCheck");
```

Comment:
please remove this code and please do in your local chekcstyle
`git checkout chekcstyle-8.35`, both plugins are behind in releases, so 8.35 do have new modules that are not present in  old metadata of plugins.